includes/wikia/tasks/Tasks/FacebookTask.class.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/**
+ * FacebookTask - Contains tasks related to Facebook
+ */
+
+namespace Wikia\Tasks\Tasks;
+
+class FacebookTask extends BaseTask {
+
+	/**
+	 * Update User email to become Facebook-reported email
+	 *
+	 * @param int $userId
+	 * @return bool
+	 */
+	public function updateEmailFromFacebook( $userId ) {
+		$userMap = \FacebookMapModel::lookupFromWikiaID( $userId );
+		if ( !$userMap ) {
+			$this->error( 'Facebook user email update fail. Missing user mapping', [
+				'title' => __METHOD__,
+				'userid' => $userId,
+			] );
+			return false;
+		}
+
+		$fbUserId = $userMap->getFacebookUserId();
+		$email = \FacebookClient::getInstance()->getEmail( $fbUserId );
+
+		if ( !$email ) {
+			$this->info( 'Facebook user email update: No Facebook email', [
+				'title' => __METHOD__,
+				'userid' => $userId,
+				'facebookid' => $fbUserId,
+			] );
+			return false;
+		}
+
+		$status = ( bool ) $this->updateUserEmail( $userMap->getWikiaUserId(), $email );
+		$this->info( 'Facebook user email update complete', [
+			'title' => __METHOD__,
+			'userid' => $userId,
+			'facebookid' => $fbUserId,
+			'email' => $email,
+			'status' => ( $status ? 'success' : 'fail' ),
+		] );
+
+		return $status;
+	}
+
+	protected function updateUserEmail( $userId, $email ) {
+		$app = \F::app();
+
+		$dbw = wfGetDB( DB_MASTER, null, $app->wg->ExternalSharedDB );
+		( new \WikiaSQL() )
```

Comment:
It looks like the user object already has a method for updating emails. [User->setEmail()](https://github.com/Wikia/app/blob/dev/includes/User.php#L2246). We should probably use this existing method since it runs a set of hooks that could do additional things like clearing caches, etc. It also keeps all the logic for updating emails in one place if that ever gets changed.